tests/functional/src/main/java/org/mule/tck/listener/ConnectionListener.java

```diff
@@ -93,4 +110,13 @@ public ConnectionListener reset()
         return this;
     }
 
+    /**
+     * @return the minimum time tracked between all received notifications
+     * @throws IllegalStateException if there were less than two notifications received
+     */
+    public Long getMinimumTimeBetweenNotifications()
```

Comment:
Is this method used anywhere else other than the test? If so, this precondition could be an assertion there.